website/www/site/content/en/documentation/ml/online-clustering.md

```diff
@@ -0,0 +1,211 @@
+
+To have a better understanding of the dataset, here are some examples from the train split of the dataset:
+
+
+| Text        | Type of emotion |
+| :---        |    :----:   |
+| im grabbing a minute to post i feel greedy wrong      | Anger       |
+| i am ever feeling nostalgic about the fireplace i will know that it is still on the property   | Love        |
+| ive been taking or milligrams or times recommended amount and ive fallen asleep a lot faster but i also feel like so funny | Fear |
+| on a boat trip to denmark | Joy |
+| i feel you know basically like a fake in the realm of science fiction | Sadness |
+| i began having them several times a week feeling tortured by the hallucinations moving people and figures sounds and vibrations | Fear |
+
+### Clustering Algorithm
+For the clustering of tweets, we use an incremental clustering algorithm called BIRCH. It stands for balanced iterative reducing and clustering using hierarchies and is an unsupervised data mining algorithm used to perform hierarchical clustering over particularly large data-sets. An advantage of BIRCH is its ability to incrementally and dynamically cluster incoming, multi-dimensional metric data points in an attempt to produce the best quality clustering for a given set of resources (memory and time constraints).
+
+
+## Ingestion to PubSub
+We first ingest the data into [PubSub](https://cloud.google.com/pubsub/docs/overview) so that while clustering we can read the tweets from PubSub. PubSub is a messaging service for exchanging event data among applications and services. It is used for streaming analytics and data integration pipelines to ingest and distribute data.
+
+The full example code for ingesting data to PubSub can be found [here](sdks/python/apache_beam/examples/inference/online_clustering/write_data_to_pubsub_pipeline/)
+
+The file structure for ingestion pipeline is:
+
+    write_data_to_pubsub_pipeline/
+    ├── pipeline/
+    │   ├── __init__.py
+    │   ├── options.py
+    │   └── utils.py
+    ├── __init__.py
+    ├── config.py
+    ├── main.py
+    └── setup.py
+
+`pipeline/utils.py` contains the code for loading the emotion dataset and two `beam.DoFn` that are used for data transformation
+`pipeline/options.py` contains the pipeline options to configure the Dataflow pipeline
+`config.py` defines some variables like GCP PROJECT_ID, NUM_WORKERS that are used multiple times
+`setup.py` defines the packages/requirements for the pipeline to run
+`main.py` contains the pipeline code and some additional function used for running the pipeline
```

Comment:
```suggestion
`pipeline/utils.py` contains the code for loading the emotion dataset and two `beam.DoFn` that are used for data transformation.

`pipeline/options.py` contains the pipeline options to configure the Dataflow pipeline.

`config.py` defines some variables like GCP PROJECT_ID, NUM_WORKERS that are used multiple times.

`setup.py` defines the packages/requirements for the pipeline to run.

`main.py` contains the pipeline code and some additional function used for running the pipeline.
```

I like the spacing in the Readme, but in practice this renders as a single sentence if you don't double space - http://apache-beam-website-pull-requests.storage.googleapis.com/23289/documentation/ml/online-clustering/index.html#ingestion-to-pubsub